python-sdk/src/astro/databases/snowflake.py

```diff
@@ -475,35 +478,62 @@ def create_table_using_native_schema_autodetection(
             },
         )
 
+    @staticmethod
+    def get_cols_case(cols: tuple[str]):
+        result = []
+        for col in cols:
+            if col.isupper():
+                result.append(1)
+            elif col.islower():
+                result.append(0)
+            else:
+                return "mixed"
+
+        total = sum(result)
+        if total == 0:
+            return "lower"
+        elif total == len(cols):
+            return "upper"
+
+    @staticmethod
+    def use_quotes(cols: tuple[str]):
+        return SnowflakeDatabase.get_cols_case(cols) in ["upper", "mixed"]
+
     def create_table_using_schema_autodetection(
         self,
         table: BaseTable,
         file: File | None = None,
         dataframe: pd.DataFrame | None = None,
-        columns_names_capitalization: ColumnCapitalization = "lower",
+        columns_names_capitalization: ColumnCapitalization = "original",
```

Comment:
This parameter isn't used anymore, right?

Reply:
Yes, we don't need the `columns_names_capitalization` feature itself. We can remove it altogether.

Reply:
We should call it out as comment as TODO

Reply:
Added a ticket - https://github.com/astronomer/astro-sdk/issues/1099